base/collect_database.go

```diff
@@ -0,0 +1,315 @@
+package base
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/itchyny/gojq"
+	log "github.com/iter8-tools/iter8/base/log"
+
+	"text/template"
+	"time"
+
+	"sigs.k8s.io/yaml"
+)
+
+type CollectDatabaseTemplateInput struct {
+	MonitoringEndpoint string `json:"monitoringEndpoint" yaml:"monitoringEndpoint"`
+	IAMToken           string `json:"IAMToken" yaml:"IAMToken"`
+	GUID               string `json:"GUID" yaml:"GUID"`
+}
+
+type CollectDatabaseTemplate struct {
+	Url      string            `json:"url" yaml:"url"`
+	Headers  map[string]string `json:"headers" yaml:"headers"` // TODO: should this be map[string]interface{} instead?
+	Provider string            `json:"provider" yaml:"provider"`
+	Method   string            `json:"method" yaml:"method"`
+	Metrics  []Metric          `json:"metrics" yaml:"metrics"`
+}
+
+type Metric struct {
```

Comment:
Shouldn't there also be a `Body string` field for POST queries?

Reply:
Added.